### DISCUSSION TITLE
Erratic Behavior with Circuit Groups

### DISCUSSION THREAD

**Discussion by rock-crusher** [tagyoureit/nodejs-poolController#784] [Nixie]
I’m experimenting with the circuit groups, but may not have this set up correctly.

I would like to find a way to get this to be accurate and dependable each time the feature button is pressed. As of now, it is a crap shoot as to how it will respond for the first couple tries. After that it will usually stabilize and will work as expected then you can press the feature button as many times thereafter and it will work as it should. Now, when it sits for approximately 10-15 minutes without trying it and you press the feature button again it’s as though it lost its memory and forgot what it is supposed to turn on and when. It will just simply turn on whatever (the crap shoot) and the cycle above then repeats itself. 

Below is how I have it set up.

<img width="313" alt="circuit groups" src="https://github.com/tagyoureit/nodejs-poolController/assets/94593954/0977581e-9b2d-43f6-89f1-6d61aacdd7c6">

The intent for this is when the “enjoy spa night” feature button is pressed it turns on the spillway for 3min, or whatever time length I decide to set it for, in this case it’s the 3 min. When activated, the spillway turns on along with the pool/spa light & the elephant water fill feature. This will allow the spa to completely fill and also skim whatever leaves/debris happens to be on the surface of the water pushing it over and through the spillway for the skimmer to get later. After the “enjoy spa night” 3 minute times out, the pool/spa light shut off, the spa fully turns on along with the tree lighting.

I’ve used this set-up before and it’s nice especially when children are playing in the spa splashing out lots of water, getting in/out and tracking all sorts of misc. debris in with them. Then, when it’s time for the adults to get in I just simply hit the “enjoy spa”, or the “enjoy spa night” feature button and the water is replenished, the top skimmed off and ready to go. 

When done, I just press the “enjoy spa off” button and everything that was on shuts off. 

Why do you think this is not working first time every time?

**Comment by rstrouse** [comment]
I think the problem is that while you can ignore a lamp post you cannot ignore an elephant.  Just kidding.  When I get a chance I will take a look.  Keep in mind the left state is what happens when the group is turned on and the right state is what happens when the group is turned off.  So Off/Ignore turns off the circuit when the group is activated and ignores it when you turn it off.   So each group has two states.

Nixie also evaluates the group on each circuit/feature change to ensure the state is as you would expect.  The moral of the story here is that these are not momentary actions, they actually have a state and can be scheduled and stuff.

**Reply by rock-crusher** [reply]
Thanks rstrouse, I'll be looking forward to what you may find once you have time to look into it.
Thanks again, RC.
